### PR TITLE
feat(sg) add grouping option for filtering security groups (closes #578)

### LINKED ISSUE #578: Add grouping by Ingress/Egress and by protocols to SGs view mode
Add grouping by Ingress/Egress and by protocols TCP/UDP/ICMP to Security group View mode. 
Filtering will be done in #506.

This task is a separate one because it is blocked by #7  (possibility to switch to the list view)

### PR DESCRIPTION
**Pull request by qqcky**

### CODE REVIEW

File: src/app/security-group/sg-rules/sg-rules.component.html [hunk 1]
```diff
@@ -154,70 +154,78 @@
...
+          <mat-option *ngFor="let z of protocols" [value]="z.value">
+            {{ z.text | translate }}
+          </mat-option>
+        </mat-select>
+      </mat-form-field>
+
+      <cs-divider-vertical></cs-divider-vertical>
+
+      <mat-form-field class="form-select-control">
+        <mat-select
```
Comment (zolotyx) [inline src/app/security-group/sg-rules/sg-rules.component.html]: use draggable select here

File: src/app/security-group/sg-rules/sg-rules.component.scss [hunk 2]
```diff
@@ -39,6 +39,10 @@ mat-form-field {
   text-align: left;
 }
 
+:host /deep/ .delete-button-cell {
```
Comment (zolotyx) [inline src/app/security-group/sg-rules/sg-rules.component.scss]: fix padding here to make it look better
fixpadding

File: src/app/security-group/sg-rules/sg-rules.component.scss [hunk 3]
```diff
@@ -58,3 +62,8 @@ mat-form-field {
 mat-form-field {
   display: inline-block !important;
 }
+
+:host cs-grouped-list .mat-list {
```
Comment (zolotyx) [inline src/app/security-group/sg-rules/sg-rules.component.scss]: do you need :host here?

File: src/app/shared/components/grouped-list/grouped-list.component.ts [hunk 4]
```diff
@@ -42,5 +42,7 @@ export class GroupedListComponent implements OnChanges {
     } else {
       this.tree = [{ items: this.list }];
     }
+
+    console.log('TREE', this.tree);
```
Comment (zolotyx) [inline src/app/shared/components/grouped-list/grouped-list.component.ts]: please remove this

File: src/app/security-group/sg-rules/sg-rules.component.ts [hunk 5]
```diff
@@ -68,6 +87,7 @@ export class SgRulesComponent {
     private dialogService: DialogService,
     private router: Router
   ) {
+    console.log(data.entity);
```
Comment (zolotyx) [inline src/app/security-group/sg-rules/sg-rules.component.ts]: please, remove this

File: src/app/security-group/sg-rules/sg-rule.component.ts [hunk 6]
```diff
@@ -1,18 +1,30 @@
...
+import {
+  GetICMPCodeTranslationToken,
+  GetICMPTypeTranslationToken
+} from '../../shared/icmp/icmp-types';
 
 @Component({
   selector: 'cs-security-group-rule',
   templateUrl: 'sg-rule.component.html',
-  styles: [`:host { display: table-row }`],
+  styles: [
```
Comment (zolotyx) [inline src/app/security-group/sg-rules/sg-rule.component.ts]: strange. but if it's really required. leave it

File: src/app/security-group/sg-rules/sg-rules.component.ts [hunk 7]
```diff
@@ -116,8 +142,11 @@ export class SgRulesComponent {
     this.networkRuleService.addRule(type, params)
       .subscribe(
         rule => {
+          console.log(rule);
```
Comment (zolotyx) [inline src/app/security-group/sg-rules/sg-rules.component.ts]: remove this

**Comment by qqcky** [comment]
Will be merged with #632

**Comment by zolotyx** [comment]
@qqcky , let's merge it to 632-sg-ngrx-store branch and verify all the functionality there

**Comment by qqcky** [comment]
Will be merged with #632 (PR #658)

**Pull request closed without merging**